### PR TITLE
[googledatastore] Update googledatastore API version to v1beta3.

### PR DESCRIPTION
**Pull request by yuyantingzero**

### CODE REVIEW

**Comment by yuyantingzero** [comment]
@stfeng , ptal

**Comment by stfeng** [comment]
Ack, I will look at this.

File: googledatastore/pom.xml [hunk 1]
```diff
@@ -29,11 +29,26 @@ LICENSE file.
   <name>Google Cloud Datastore Binding</name>
   <url>https://github.com/GoogleCloudPlatform/google-cloud-datastore</url>
 
+  <repositories>
+    <repository>
+      <id>sonatype-snapshots</id>
+      <name>sonatype-snapshots</name>
+      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
+      <snapshots>
+	<enabled>true</enabled>
```
Comment (stfeng) [inline googledatastore/pom.xml]: indentation.

File: googledatastore/pom.xml [hunk 2]
```diff
@@ -47,4 +62,28 @@ LICENSE file.
       <scope>provided</scope>
     </dependency>
   </dependencies>
+  <build>
```
Comment (stfeng) [inline googledatastore/pom.xml]: why is this one showing up here in this PR? You might want to do a rebase again?

File: googledatastore/src/main/java/com/yahoo/ycsb/db/GoogleDatastoreClient.java [hunk 3]
```diff
@@ -243,7 +242,6 @@ public Status scan(String table, String startkey, int recordcount,
   @Override
   public Status update(String table, String key,
       HashMap<String, ByteIterator> values) {
-
     return doSingleItemMutation(table, key, values, MutationType.UPDATE);
```
Comment (stfeng) [inline googledatastore/src/main/java/com/yahoo/ycsb/db/GoogleDatastoreClient.java]: One empty line is missing here in the new code. 

There has been a reformatting effort recently to all the code, please make sure this PR retains the outcome of that reformatting effort.

**Comment by yuyantingzero** [comment]
Addressed all comments, @stfeng

File: googledatastore/README.md [hunk 4]
```diff
@@ -20,6 +20,9 @@ LICENSE file.
 
 https://cloud.google.com/datastore/docs/concepts/overview?hl=en
 
+Please refer to https://cloud.google.com/datastore/docs/apis/overview for more information on
```
Comment (stfeng) [inline googledatastore/README.md]: Consider replace the link above with this new link here. And doing that, you can remove this "please refer to..." paragraph.

**Comment by stfeng** [comment]
LGTM. One minor comment about a link in the README.md file, please address.

Otherwise, the code is ready for merge, but I will let this PR sit for another couple of days in case anyone else wanted to take a look as well.

Thanks for doing this Yanting!

File: googledatastore/README.md (hunk 4 above)
Comment (yuyantingzero) [inline googledatastore/README.md]: The old link is an overview about Google Cloud Datastore. The new link contains only v1beta3 API documentation.

File: googledatastore/pom.xml [hunk 5]
```diff
@@ -29,11 +29,26 @@ LICENSE file.
...
+    </repository>
+  </repositories>
   <dependencies>
     <dependency>
-      <groupId>com.google.apis</groupId>
-      <artifactId>google-api-services-datastore-protobuf</artifactId>
-      <version>v1beta2-rev1-3.0.2</version>
+      <groupId>com.google.cloud.datastore</groupId>
+      <artifactId>datastore-v1beta3-proto-client</artifactId>
+      <version>0.0.1-SNAPSHOT</version>
```
Comment (risdenk) [inline googledatastore/pom.xml]: Do we have to depend on SNAPSHOT releases? @busbey was this something that was talked about for another binding at one point?

Comment (busbey) [inline googledatastore/pom.xml]: yeah, no SNAPSHOT dependencies please. They're a big no-no for stable maven builds.

@yuyantingzero or @stfeng could y'all get whomever publishes the datastore artifacts to make a release? (alpha / beta releases are fine so long as they're subject to the normal maven repository rules around non-SNAPSHOT versions.)

Comment (yuyantingzero) [inline googledatastore/pom.xml]: @busbey @risdenk , thanks for the review! Sounds good to me. I will ping you after updating the source.

**Comment by yuyantingzero** [comment]
@busbey @risdenk Replaced with beta releases.

File: googledatastore/pom.xml [hunk 6]
```diff
@@ -29,11 +29,21 @@ LICENSE file.
   <name>Google Cloud Datastore Binding</name>
   <url>https://github.com/GoogleCloudPlatform/google-cloud-datastore</url>
 
+  <repositories>
+    <repository>
+      <id>sonatype-snapshots</id>
+      <name>sonatype-snapshots</name>
+      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
```
Comment (busbey) [inline googledatastore/pom.xml]: do we still need this snapshot repository? I think the client is in maven central.

File: googledatastore/pom.xml [hunk 7]
```diff
@@ -29,11 +29,21 @@ LICENSE file.
...
+    </repository>
+  </repositories>
   <dependencies>
     <dependency>
-      <groupId>com.google.apis</groupId>
-      <artifactId>google-api-services-datastore-protobuf</artifactId>
-      <version>v1beta2-rev1-3.0.2</version>
+      <groupId>com.google.cloud.datastore</groupId>
+      <artifactId>datastore-v1beta3-proto-client</artifactId>
+      <version>1.0.0-beta</version>
```
Comment (busbey) [inline googledatastore/pom.xml]: this version came out on 30 Mar. It looks like a 1.0.0-beta.1 came out on 11 Apr, should we bump to that one so long as we're updating?

**Comment by yuyantingzero** [comment]
@busbey , Thanks for the review. PTAL

**Pull request merged**